static/js/actions/image_upload.js

```diff
@@ -0,0 +1,35 @@
+// @flow
+import { createAction } from 'redux-actions';
+
+import { updateProfileImage } from '../util/api';
+
+export const START_PHOTO_EDIT = 'START_PHOTO_EDIT';
+export const startPhotoEdit = createAction(START_PHOTO_EDIT);
+
+export const CLEAR_PHOTO_EDIT = 'CLEAR_PHOTO_EDIT';
+export const clearPhotoEdit = createAction(CLEAR_PHOTO_EDIT);
+
+export const UPDATE_PHOTO_EDIT = 'UPDATE_PHOTO_EDIT';
+export const updatePhotoEdit = createAction(UPDATE_PHOTO_EDIT);
+
+export const REQUEST_PATCH_USER_PHOTO = 'REQUEST_PATCH_USER_PHOTO';
+export const requestPatchUserPhoto = createAction(REQUEST_PATCH_USER_PHOTO);
+
+export const RECEIVE_PATCH_USER_PHOTO_FAILURE = 'RECEIVE_PATCH_USER_PHOTO_FAILURE';
+export const receivePatchUserPhotoFailure = createAction(RECEIVE_PATCH_USER_PHOTO_FAILURE);
+
+export const RECEIVE_PATCH_USER_PHOTO_SUCCESS = 'RECEIVE_PATCH_USER_PHOTO_SUCCESS';
+export const receivePatchUserPhotoSuccess = createAction(RECEIVE_PATCH_USER_PHOTO_SUCCESS);
+
+export function updateUserPhoto(username: string, image: Blob, name: string): Dispatcher<string> {
+  return (dispatch: Dispatch) => {
+    dispatch(requestPatchUserPhoto(username));
+    return updateProfileImage(username, image, name).then(
+      () => {
+        dispatch(receivePatchUserPhotoSuccess(username));
+      },
+      error => {
+        dispatch(receivePatchUserPhotoFailure(username, error));
```

Comment:
actually, do you think it's necessary to do anything with the username here? User's aren't going to be able to update profile images for other users.